ratelimits/source_redis.go

```diff
@@ -125,7 +105,10 @@ func (r *RedisSource) Get(ctx context.Context, bucketKey string) (time.Time, err
 		}
 
 		state := "failed"
-		if errors.Is(err, context.DeadlineExceeded) {
+		var netErr net.Error
+		if errors.As(err, &netErr) && netErr.Timeout() {
```

Comment:
I assume the `netErr.Timeout` case was seen when attempting to dial an unresponsive redis? If that's the case, should this also be used for `Set` and `Delete` operations?

Reply:
I'm going off of the fact that go-redis uses the `net.Dialer` to build a pool of tcp `net.Conn`. When using a `net.Conn` from the pool it sets a deadline on the context passed into `Get()`/`Set()`/`Delete()`.

https://github.com/redis/go-redis/blob/c3098d5f7eba88330638a8b2b3e23d42cd5c9338/internal/pool/conn.go#L70

https://github.com/redis/go-redis/blob/c3098d5f7eba88330638a8b2b3e23d42cd5c9338/internal/pool/conn.go#L81

Because of that, and combined with the error cases that they document (e.g. `redis.Nil`) the error cases below are the ones that I actually expect would be returned by the client.

Also, you're right, I forgot to include this addition elsewhere.

Reply:
Alright I've moved the error classification out of each method into a utility function and beefed it up a bit.